### ISSUE TITLE
LSP always attempts to start a server when there isn't one running

### ISSUE THREAD

**Issue by nigredo-tori** [syl20bnr/spacemacs#15334]
Consider this line: https://github.com/syl20bnr/spacemacs/blob/efdfecbf8f72d3de4a410f9d653f8b91e569063c/layers/%2Blang/scala/funcs.el#L28

To understand why this is problematic, we need to know the difference between the `lsp` and `lsp-mode` commands:
1. `lsp-mode` is a minor mode that sets keymaps, window decoration, and so on. It, however, does not launch any processes. (**UPD:** this is not correct - see the comment below) ~~It attempts to locate a running LSP server associated with the current project/language, and tries to connect to it. However, if there is no such server to connect to, `lsp-mode` just shrugs and does nothing, except for maybe displaying the connection state in the mode line.~~ In such a state the LSP keymap is still active - so if the user so chooses, they can do `, w s` (bound to `lsp` in `scala-mode`, YMMV) to actually start an LSP server.
2. `lsp` is a command that **explicitly** launches an LSP server if one is not running, asking the user for input where it is needed. It does enable `lsp-mode` so you still get the keymap, but that is more of an afterthought.

Returning to the line in question, we can see that it uses `lsp` rather than `lsp-mode`. Which means that, whenever I open a Scala file in a project where I don't have a server running, the following happens.
1. A Metals (Scala LSP server) process is started, resulting in a minor freeze.
2. If the project was not imported (scala-specific process, not sure if other LSP servers have something like that), I get a prompt to import it.
3. If the project was imported, but its settings have since changed (e.g. because of a checkout), then I get another prompt to reimport it.
4. If I chose to import the project for some reason, I might get yet another prompt that proposes I look at import errors.
4. Metals attempts to compile a part of the project, which can mean a significant CPU and memory load.
5. If I don't want my resources going to waste, I can stop Metals (`, w q`).
6. Finally after all this interruption I can go back to work.

If I did stop Metals, then whenever I open another file from this project I have to go through all of this again! This gets extremely annoying if I just want to look at the code without a full-blown IDE experience.
  
What I propose we do instead is to use `lsp-mode` (**UPD:** `lsp-mode` doesn't work this way - see comment below) instead of `lsp` for hooks by default (so that it starts disconnected), and make the user launch their LSP server explicitly by doing `, w s` (`lsp`). Users that prefer the current "always on" behavior can get it by adding `lsp` to their major mode hooks in their `.spacemacs`. Or we can add a layer variable or a customization option to switch between these two behaviors.

**Comment by nigredo-tori** [comment]
Apparently, I got `lsp-mode` wrong, and it **does not** attempt to connect to running servers. Meaning that what I proposed becomes more difficult, and will probably require upstream changes in `lsp-mode`. At least it seems this way, since I can't find an existing way to connect to running LSP servers without defaulting to a fresh one.

**Comment by lebensterben** [comment]
@nigredo-tori Feel free to update your earlier comments. In current state it's a bit hard to follow what's the actual behaviour...
